### ISSUE TITLE
Update category synch documentation

### ISSUE THREAD

**Issue by katmatt** [commercetools/commercetools-sync-java#243]
The documentation for [commercetools category sync](https://github.com/commercetools/commercetools-sync-java/blob/master/docs/usage/CATEGORY_SYNC.md) isn't up to date, the code snippet:
```
final Logger logger = LoggerFactory.getLogger(MySync.class);
final CategorySyncOptions categorySyncOptions = CategorySyncOptionsBuilder.of(sphereClient)
                                                                          .setErrorCallBack(logger::error)
                                                                          .setWarningCallBack(logger::warn)
                                                                          .build();
```
doesn't compile anymore. 

You changed the names of the methods (and their signature) - which is a great idea 👍  But it would be nice if you could update the documentation 😄 

And I guess that this inconsistency exists in the documentation for the products and inventory entries syncher doc too 😉

**Comment by heshamMassoud** [comment]
@katmatt Thanks for reporting this. :) Will fix it asap.

**Comment by heshamMassoud** [comment]
Fixed in #244
